recipes/hipace/meta.yaml

```
@@ -0,0 +1,93 @@
+{% set name = "hipace" %}
+{% set version = "23.02" %}
+{% set build = 0 %}
+{% set sha256 = "f4f5958f0600dd59cc1bb02218d5b7d1e6e0da34461c504e4d0367c98d5d6113" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Hi-PACE/hipace/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: {{ build }}
+  run_exports:
+    # releases are not (yet) compatible
+    - hipace {{ version }}
+
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - ccache        # [unix]
+    - clang         # [win]
+    - cmake >=3.20.0
+    - lld           # [win]
+    # TODO: add libxml2 dep in lld-feedstock
+    - libxml2       # [win]
+    - make          # [unix]
+    - ninja         # [win]
+    - libgomp       # [linux]
+    - llvm-openmp   # [osx or win]
+  host:
+    - fftw
+    - openpmd-api >=0.14.5
```

This triggers the need for a newer numpy, that for some reason does not exist in the `staged-recipes` repo...?

---

HiPACE++ now requires `openpmd-api >=0.15.0`

---

Thanks!

To be precise, now as of 23.04+, currently working on packaging 23.03+.

---

Let's just take the latest:
```suggestion
    - openpmd-api
```

---

Urgh, does not do the trick and tries to pull in 0.14.4 o.0

---

@mfansler do/can we use `libmamba` or `mamba` here? The dependency resolver does not find the latest versions (0.15.1).

---

Sorry, just saw this. Conda Forge already uses `mamba` as the solver for all CI ops. The situation looks like `numpy=1.24` is getting used in recent `openpmd-api` builds. But currently using plain `numpy` here gets replaced with `numpy=1.21`, hence only older `openpmd-api` versions get resolved.

In local solving tests (not building) I was able to get `openpmd-api >=0.14.5` to work with setting `numpy=1.23`. You can either try:

 - setting a stricter bound like this on `numpy` in this recipe; or 
 - loosen the `openpmd-api` builds to let it use the `numpy` global pinning. 

I think the latter could be achieved by removing [the explicit bound on `numpy` (`>=1.15.0`)](https://github.com/conda-forge/openpmd-api-feedstock/blob/d4bb38a4b402c533b11287a128a582f3c0cf95e9/recipe/meta.yaml#L73), which I suspect is what is interfering with the default global pinning (`numpy=1.21`).

Maybe someone from @conda-forge/help-python could weigh in on this. I don't want to recommend rebuilding that feedstock unnecessarily, since it has so many variants. 😬 But I believe getting the `openpmd-api` builds compatible with global pinning would be preferable.

---

Please see https://github.com/conda-forge/openpmd-api-feedstock/pull/102#issuecomment-1539217537 for a concrete example about what our docs say regarding pinning numpy versions. Consider rebuilding necessary if it prevents unsolvable environments.

---

Also, it's not uncommon to need to rebuild packages with newer versions of their dependencies.

---

Thanks a lot for the guidance - fix coming in https://github.com/conda-forge/openpmd-api-feedstock/pull/103